recipes/ncurses/all/conanfile.py

```diff
@@ -41,90 +49,86 @@ class NCursesConan(ConanFile):
         "with_pcre2": False,
     }
 
-    generators = "pkg_config"
-    exports_sources = "patches/*"
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
     @property
     def _settings_build(self):
         return getattr(self, "settings_build", self.settings)
 
     @property
-    def _with_ticlib(self):
-        if self.options.with_ticlib == "auto":
-            return self.settings.os != "Windows"
-        else:
-            return self.options.with_ticlib
+    def _is_mingw(self):
+        return self.settings.os == "Windows" and self.settings.compiler == "gcc"
 
-    @property
-    def _with_tinfo(self):
-        if self.options.with_tinfo == "auto":
-            return self.settings.os != "Windows"
-        else:
-            return self.options.with_tinfo
+    def export_sources(self):
+        export_conandata_patches(self)
+        copy(self, "*.cmake", src=self.recipe_folder, dst=self.export_sources_folder)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
+        if self.options.with_ticlib == "auto":
+            self.options.with_ticlib = self.settings.os != "Windows"
+        if self.options.with_tinfo == "auto":
+            self.options.with_tinfo = self.settings.os != "Windows"
```

Comment:
```suggestion
        if self.options.with_ticlib == "auto":
            self.options.with_ticlib = self.settings.os != "Windows"
        if self.options.with_tinfo == "auto":
            self.options.with_tinfo = self.settings.os != "Windows"
```

Need to move to configure() method. config options can not acquire options values passed by users.

Reply:
Hmm, I was getting my hopes up for this working based on https://github.com/conan-io/conan-center-index/blob/71b82701d768a332ea97a9874a1fd01942ec1440/recipes/arrow/all/conanfile.py#L161-L175
Tested it again locally just to be sure and it indeed does not work. The linked logic in arrow is totally broken then, I guess?